vscode/quint-vscode/scripts/prepare-release.sh

```diff
@@ -1,51 +1,97 @@
+SERVER_DIR="$PLUGIN_DIR"/server
+PLUGIN_CHANGELOG="$PLUGIN_DIR"/CHANGELOG.md
+
+if [[ -z "${1-}" ]]
+then
+    echo "ERROR: An argument valid for 'npm version <arg>' must be supplied."
+    echo "See 'npm version --help'."
+    exit 1
+fi
+bump_increment="$1"
 
 git checkout main;
 git pull origin main;
 
-echo "Ensure that the quint dependency is correct, currently: $QUINT_VERSION. Set the QUINT_VERSION environment variable to update it."
-pause 'Press [Enter] key to continue...'
-
-echo "Ensure that the VSCode version was bumped."
-echo "Extension: $(cat package.json | jq .version | xargs)."
-echo "Server: $(cat server/package.json | jq .version | xargs)."
-echo "You can run npm version <major|minor|patch> to bump the correct level. Remember to bump the server as well."
-pause 'Press [Enter] key to continue...'
-
-echo "Ensure that CHANGELOG.md is updated: There is an entry for this release and no UNRELEASED header."
-pause 'Press [Enter] key to continue...'
-
-cd server;
-yalc remove @informalsystems/quint;
-npm install @informalsystems/quint@$QUINT_VERSION;
-cd ..;
-npm install;
-npm run compile;
-
-version=$(cat package.json | jq .version | xargs)
-git checkout -b vscode-release/$version;
-git add CHANGELOG.md;
-git add package.json;
-git add package-lock.json;
-git add server/package.json;
-git add server/package-lock.json;
-git commit -S -m "Release VSCode extension v$version"
-
-echo "All good!"
-echo "Now check that everything is alright, push, and open a PR."
-echo "When the PR is merged, run vsce publish on an updated main branch."
+QUINT_VERSION=$(cd "$QUINT_DIR" && npm pkg get version | sed 's/"//g')
+
+# See https://stackoverflow.com/a/1885534/1187277
+read -p "Prepare [${bump_increment}] release for vscode plugin targeting quint version [$QUINT_VERSION]? [yY]" -n 1 -r
+echo
+if ! [[ $REPLY =~ ^[Yy]$ ]]
+then
+    echo "Canceling release"
+    exit 1
+fi
+
+# Bump plugin version
+(cd "$PLUGIN_DIR" && npm version "$bump_increment")
+(cd "$SERVER_DIR" && npm version "$bump_increment")
+version=$(cd "$PLUGIN_DIR" && npm pkg get version | sed 's/"//g')
+
+# Update changelog
+TMP_CHANGES=$(mktemp)
+trap 'rm "$TMP_CHANGES"' EXIT
+
+(
+cd "$PLUGIN_DIR"
+release_date=$(printf '%(%Y-%m-%d)T\n' -1)
+version_heading="## v${version} -- ${release_date}"
+
+# Update the changelog
+sed -i "s/## UNRELEASED/## UNRELEASED\n\
+\n\
+### Added\n\
+### Changed\n\
+### Deprecated\n\
+### Removed\n\
+### Fixed\n\
+### Security\n\
+\n\
+${version_heading}/" "$PLUGIN_CHANGELOG"
+)
+
+# Get the release notes for the version by printing all lines between lines
+# starting with the version header and ending with the next version header,
+# inclusive. We then drop the last line to remove the subsequent version header.
+# It's saved to a temp file for later use.
+sed -n "/## v$version/,/## v/p" "$PLUGIN_CHANGELOG" | head -n -1 > "$TMP_CHANGES"
+
+# Ensure we can build package?
+# TODO: I'm not sure why we build/install here
```

Comment:
The install might be redundant, I'm not sure. But `npm run compile` is to make sure that everything works fine with the installed version of the quint dependency. Perhaps it would make sense to run the tests as well, now that we have some.

Reply:
I think we’re ok on the tests, since those should run in CI for the release candidate PR?